osu.Game/Online/Leaderboards/LeaderboardScore.cs

```diff
@@ -377,6 +388,36 @@ public RankLabel(int? rank)
             public LocalisableString TooltipText { get; }
         }
 
+        private class DateLabel : DrawableDate
+        {
+            public DateLabel(DateTimeOffset date)
+                : base(date, 20)
+            {
+                Anchor = Anchor.Centre;
+                Origin = Anchor.Centre;
+            }
+
+            protected override string Format()
+            {
+                var now = DateTime.Now;
+                var difference = now - Date;
+
+                // TODO(optional): support localisation (probably via `CommonStrings.CountHours()` etc.)
+                // requires pluralisable string support framework-side
+
+                if (difference.TotalMinutes < 1)
+                    return CommonStrings.TimeNow.ToString();
+                if (difference.TotalHours < 1)
+                    return $@"{Math.Ceiling(difference.TotalMinutes)}min";
+                if (difference.TotalDays < 1)
+                    return $@"{Math.Ceiling(difference.TotalHours)}h";
+                if (difference.TotalDays < 7)
+                    return $@"{Math.Ceiling(difference.TotalDays)}d";
```

Comment:
Also I'll note that the [rankings table already does something similar](https://github.com/ppy/osu/blob/34135138b66a19ce00e676df7c83361ad4154304/osu.Game/Overlays/BeatmapSet/Scores/ScoreboardTime.cs) and I'd rather see that split out to a common place rather than have another local implementation. Especially so that the formats don't match (and they don't match on purpose, as the web suffixes are sort of confusing).